src/core/dom/style.js

```diff
@@ -226,35 +230,35 @@ export function deg(value) {
   return `${value}deg`;
 }
 
+/**
+ * Coerces a number into a string with units.
+ * @param {number|string} value
+ * @param {function(number):string} fn
+ * @return {string}
+ */
+function units(value, fn) {
+  return typeof value == 'number' ? fn(value) : value;
+}
+
 /**
  * Returns a "translateX" for CSS "transform" property.
  * @param {number|string} value
  * @return {string}
  */
 export function translateX(value) {
-  if (typeof value == 'string') {
-    return `translateX(${value})`;
-  }
-  return `translateX(${px(value)})`;
+  return `translateX(${units(value, px)})`;
 }
 
 /**
  * Returns a "translateX" for CSS "transform" property.
  * @param {number|string} x
- * @param {(number|string)=} opt_y
+ * @param {(number|string|null)=} opt_y
  * @return {string}
  */
 export function translate(x, opt_y) {
-  if (typeof x == 'number') {
-    x = px(x);
-  }
-  if (opt_y === undefined) {
-    return `translate(${x})`;
-  }
-  if (typeof opt_y == 'number') {
-    opt_y = px(opt_y);
-  }
-  return `translate(${x}, ${opt_y})`;
+  return opt_y === undefined || opt_y === null
```

Comment:
```suggestion
  return opt_y == null
```

Reply:
This PR was on auto-submit, but I've added this change to https://github.com/ampproject/amphtml/pull/35688/commits/1dcc7001d46adcdeb17412b4c1062a770cf68958